src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesignerAssembly.cs

```diff
@@ -0,0 +1,340 @@
+			foreach (var line in lines) {
+				var items = line.Split (new char [] { ' ' }, 4);
+				int value = items [1] != "styleable" ? Convert.ToInt32 (items [3], 16) : -1;
+				string itemName = GetResourceName(items [1], items [2], resource_fixup);
+				switch (items [1]) {
+				case "anim":
+				case "animator":
+				case "attr":
+				case "array":
+				case "bool":
+				case "color":
+				case "dimen":
+				case "drawable":
+				case "font":
+				case "id":
+				case "integer":
+				case "interpolator":
+				case "layout":
+				case "menu":
+				case "mipmap":
+				case "plurals":
+				case "raw":
+				case "string":
+				case "style":
+				case "transition":
+				case "xml":
+					CreateIntProperty (items [1], itemName, value, resourceDesigner, module);
+					break;
+				case "styleable":
+					switch (items [0]) {
+					case "int":
+						CreateIntProperty (items[1], itemName, Convert.ToInt32 (items [3], 10), resourceDesigner, module);
+						break;
+					case "int[]":
+						var arrayValues = items [3].Trim (new char [] { '{', '}' })
+							.Replace (" ", "")
+							.Split (new char [] { ',' });
+						CreateIntArrayProperty (items[1], itemName,
+							arrayValues.Select (x => string.IsNullOrEmpty (x) ? -1 : Convert.ToInt32 (x, 16)).ToArray (), resourceDesigner, module);
+						break;
+					}
+					break;
+				// for custom views
+				default:
+					CreateIntProperty (items [1], itemName, value, resourceDesigner, module);
+					break;
+				}
+			}
+		}
+
+		void CreateIntProperty (string resourceClass, string propertyName, int value, TypeDefinition resourceDesigner, ModuleDefinition module)
+		{
+			TypeDefinition nestedType = CreateResourceClass (resourceDesigner, resourceClass, module);
+			PropertyDefinition p = CreateProperty (propertyName, value, module);
+			nestedType.Properties.Add (p);
+			nestedType.Methods.Insert (Math.Max(0, nestedType.Methods.Count () -1), p.GetMethod);
+		}
+
+		void CreateIntArrayProperty (string resourceClass, string propertyName, int[] values, TypeDefinition resourceDesigner, ModuleDefinition module)
+		{
+			TypeDefinition nestedType = CreateResourceClass (resourceDesigner, resourceClass, module);
+			PropertyDefinition p = CreateArrayProperty (propertyName, values, module);
+			nestedType.Properties.Add (p);
+			nestedType.Methods.Insert (Math.Max(0, nestedType.Methods.Count () -1), p.GetMethod);
+		}
+
+		Dictionary<string, TypeDefinition> resourceClasses = new Dictionary<string, TypeDefinition> ();
+
+		void CreateCtor (TypeDefinition type, ModuleDefinition module)
+		{
+			var ctor = new MethodDefinition(".ctor", MethodAttributes.SpecialName | MethodAttributes.RTSpecialName | MethodAttributes.HideBySig | MethodAttributes.Public, module.TypeSystem.Void);
+			var ctoril = ctor.Body.GetILProcessor();
+			ctoril.Emit(OpCodes.Ldarg_0);
+			var o = module.TypeSystem.Object.Resolve();
+			ctoril.Emit(OpCodes.Call, module.ImportReference(o.Methods.First(x => x.IsConstructor)));
+			ctoril.Emit(OpCodes.Nop);
```

Comment:
Why nop?

Reply:
I added that because it was in the IL code which I based this on. What I did was build an assembly normally, then look at the IL is produced and replicated that. 

If this is not needed, then I can take it out.

Reply:
@dellis1972 you might compare a `Release` assembly to see if that is any different?

Reply:
Good idea.